### PR TITLE
pythonPackages.f90nml: init at 1.4.1

### PR DESCRIPTION
**Pull request by loicreynier**
###### Description of changes

Add [`f90nml`](https://f90nml.readthedocs.io/), a Python module for working with Fortran Namelists.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/f90nml/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib, pythonPackages, setuptools-scm }:
+
+with pythonPackages;
+
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/f90nml/default.nix:5]:
```suggestion
{ lib, buildPythonPackage, setuptools-scm }:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/f90nml/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "1.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-nfMSqhO5whk28FnKuatAr+vCgPGrVOaFTDhz0Le3hlw=";
+  };
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/f90nml/default.nix:16]: There are tests https://github.com/marshallward/f90nml we want to execute and a pythonImportsCheck is missing.

Comment (loicreynier) [reply pkgs/development/python-modules/f90nml/default.nix:14]: Just added the `pythonImportsCheck` but couldn't make the tests to works. Tests are looking for test files which they cannot find.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/f90nml/default.nix:14]: You probably need to fetch from github for that to work.

Comment (loicreynier) [reply pkgs/development/python-modules/f90nml/default.nix]: Indeed the test files are not distributed on Pypi.

File: pkgs/development/python-modules/f90nml/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "marshallward";
+    repo = pname;
+    rev = "v" + version;
+    sha256 = "sha256-nSpVBAS2VvXIQwYK/qVVzEc13bicAQ+ScXpO4Rn2O+8=";
+  };
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/f90nml/default.nix:16]:
```suggestion
```
that is the default which we don't need to repeat.
I was more thinking we need something like pytest. Right now it looks like we are not executing any tests.
We should probably do something like:

```suggestion
  checkPhase = ''
    ${python.interpreter} setup.py check
  '';

```

Comment (loicreynier) [reply pkgs/development/python-modules/f90nml/default.nix]: Test are now properly executed, thanks.

**Review by jonringer — COMMENTED** [review]
Do you mind rebasing to get rid of the merge commits?

Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

 `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by loicreynier** [comment]
Commit history is now squashed in one commit.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.f90nml

**Pull request merged**